apps/dg/controllers/data_context.js

```diff
@@ -968,12 +1086,12 @@ DG.DataContext = SC.Object.extend((function() // closure
     var childCollectionID = iCollection && iCollection.get('id'),
         collectionCount = this.get('collectionCount'),
         collections = this.get('collections'),
-        collectionRecord, collectionID,
+        collection, collectionID,
         prevCollectionID = null;
     for( var i = 0; i < collectionCount; ++i) {
-      collectionRecord = collections.objectAt( i);
-      collectionID = collectionRecord.get('id');
-      if( collectionRecord && (collectionID === childCollectionID))
+      collection = collections.objectAt( i);
+      collectionID = collection.get('id');
+      if( collection && (collectionID === childCollectionID))
```

Comment:
This seems like the place where order dependency is introduced into getting the parent collection or child collection. True? If so, would now be a good time to get rid of it and do something not order dependent?